File: CS50x/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # variable declaration
    para = get_string("Text: ")
    lettersc = count_letters(para)
    wordsc = count_words(para)
    sentencesc = count_sentences(para)

    # To calculate the grade
    L = (lettersc / wordsc) * 100
    S = (sentencesc / wordsc) * 100

    # Roundingoff the grade
    graded = round(0.0588 * L - 0.296 * S - 15.8)

    if graded < 1:
        print("Before Grade 1")

    elif graded > 16:
        print("Grade 16+")

    else:
        print(f"Grade {graded}")


# Calculate the number of charaters in the text
def count_letters(text):
    counter = 0

    # Looping through the word
    for i in range(len(text)):
        if text[i].islower() or text[i].isupper():

            # Counting the character in the word
            counter += 1

    return counter


# Calculate the  number of words in the text
def count_words(text):
    counter = 0

    # Looping through the sentence in the textfile
    for i in range(len(text)):
        if text[i] == ' ':

            # counting the words in the text
            counter += 1

    return counter + 1

# Calculate the number of sentences in the text


def count_sentences(text):
    counter = 0

    # Looping through the text in the text file
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':

            # Counting the sentences in the textfile
            counter += 1

    return counter


# Main function
if __name__ == "__main__":
    main()

File: CS50x/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    int letters;
    int words;
    int sentences;
    int index;

    // prompts user for text input
    string text = get_string("Text: ");

    // count the letter
    letters = count_letters(text);
    // printf("%i letters\n", letters);

    // count the word
    words = count_words(text);
    // printf("%i words\n", words);

    // count the sentences
    sentences = count_sentences(text);
    // printf("%i sentences\n", sentences);

    float L = (float)letters / (float) words * 100;
    float S = (float)sentences / (float) words * 100;

    index = (int)round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

    return 0;
}

// count the letter
int count_letters(string text)
{
    int count = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] > 64 && text[i] < 91) || (text[i] > 96 && text[i] < 123))
        {
            count++;
        }
    }
    return count;
}

// count the words
int count_words(string text)
{
    int count = 0;
    int i = 0;

    while (text[i])
    {
        if (text[i] == ' ')
        {
            count++;
        }
        i++;
    }
    return count + 1;
}

// count the sentences
int count_sentences(string text)
{
    int count = 0;
    int i = 1;

    while (text[i])
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
        i++;
    }

    return count;
}